lib/runAll.js

```diff
@@ -107,14 +114,41 @@ export const runAll = async (
     return ctx
   }
 
-  const stagedFileChunks = chunkFiles({ baseDir: gitDir, files, maxArgLength, relative })
-  const chunkCount = stagedFileChunks.length
-  if (chunkCount > 1) debugLog(`Chunked staged files into ${chunkCount} part`, chunkCount)
+  const configDir2config = {
+    [gitDir]: config,
+  }
+  const configDir2Files = {
+    [gitDir]: [],
+  }
+  if (sparseConfig) {
+    for (const file of files) {
+      const absoluteFilepath = path.resolve(gitDir, file)
+      const subConfigInfo = await loadConfig({
+        searchStartPath: path.dirname(absoluteFilepath),
+        logger,
+      })
```

Comment:
Just a thought... using `await` inside the `for...of` loop means config files are discovered serially. I wonder if it would be possible to somehow make it parallel, for example:

```js
await Promise.all(files.map((file) => {
    const absoluteFilepath = path.resolve(gitDir, file)
    return loadConfig({
         searchStartPath: path.dirname(absoluteFilepath),
         logger,
    }).then((subConfigInfo) => {
        ....
    })
}))

Reply:
Nice catch! Let me improve this part of
 logic.